tempodb/encoding/vparquet/schema.go

```diff
@@ -619,5 +619,7 @@ func extendReuseSlice[T any](sz int, in []T) []T {
 		return in[:sz]
 	}
 
-	return make([]T, sz)
+	// append until we're large enough
+	in = in[:cap(in)]
+	return append(in, make([]T, sz-len(in))...)
```

Comment:
In the surface this seems like it would incur 2 allocations:  one in `make(...)` for the new segment, then again inside `append` because we already verified that the capacity is too small.  Is this more efficient than the original, which is 1 alloc?  Wonder if there is any compiler/runtime magic happening.

Reply:
Adding another thought:  `append` will over-allocate mem (2*size?) and that would be more efficient when growing the slices than allocing the exact size.  Perhaps `return make([]T, sz*2)` would be the best third option.

Reply:
nice catch. a benchmark is showing only one alloc:

```
func BenchmarkExtendReuseSlice(b *testing.B) {
	in := []int{1, 2, 3}
	for i := 0; i < b.N; i++ {
		_ = extendReuseSlice(100, in)
	}
}
```

```
BenchmarkExtendReuseSlice-8   	 3197289	       397.9 ns/op	     896 B/op	       1 allocs/op
```

I imagine that this pattern is common enough that the compiler optimizes it into 1 alloc. Generally I've found that all of the built in slice functions have these kinds of hidden optimizations.

Reply:
> Adding another thought: append will over-allocate mem (2*size?)

a quick test is showing worse performance with 2x here. I imagine it's overallocating quite a bit of space for huge traces which is then going unused.